iree/compiler/InputConversion/MHLO/MHLOToLinalgOnTensors.cpp

```diff
@@ -209,6 +209,49 @@ struct FftOpConversion : public OpConversionPattern<mhlo::FftOp> {
 };
 }  // namespace
 
+//===----------------------------------------------------------------------===//
+// mhlo.fft conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+// TODO(#6402): Delete the pattern after no invalid ops are created and the
```

Comment:
Is this critical to land. This seems like a WAR for a bug in lowering. Can we just fix that issue. Makes me queasy to have lowering for incorrect ops.

Reply:
This is the last piece to compile ASR frontend+encoder model. The issue is very very tricky, honestly. I already tracked it back to the start of MLIR file, all the lowering and type inference look good to me.

It is coming from:

```
%cst_127 = "tf.Const"() {device = "", value = dense<0> : tensor<1xi32>} : () -> tensor<1xi32>
%cst_128 = "tf.Const"() {device = "", value = dense<-1> : tensor<1xi32>} : () -> tensor<1xi32>
%cst_129 = "tf.Const"() {device = "", value = dense<2> : tensor<1xi32>} : () -> tensor<1xi32>
%58 = "tf.StridedSlice"(%56, %cst_127, %cst_128, %cst_129) {begin_mask = 0 : i64, device = "", ellipsis_mask = 0 : i64, end_mask = 0 : i64, new_axis_mask = 0 : i64, shrink_axis_mask = 0 : i64} : (tensor<*xi32>, tensor<1xi32>, tensor<1xi32>, tensor<1xi32>) -> tensor<*xi32>
```

And the `%56` does have shape tensor<6xi32>. However the constant operand (ie `-1`) make it infers wrong attributes. The computation of attribute is correct. I think this is coming from model configuration.

Reply:
We can hang it over a bit, I'll ask if others on model side can verify it.

Reply:
If I am reading this right, it is setting the end to -1 ? If it was meant to be dynamic or something then it has to be encoded somewhere in the op or its lowering out of `tf` dialect. If it is an input error, even less reason to have a WAR for it. Lets see what the model folks say and thanks for digging into the error.

Reply:
It will be offsetted base on the shape, ie, `-1 + size`.

Reply:
This is my calculation mistake. It's not a bug.

slice(0, 5, 2) the indices are [0, 2, 4], so the size is 3.